cmd/root.go

```diff
@@ -130,6 +143,11 @@ var (
 					logrus.WithError(err).Warning("Failed to shutdown OpenTelemetry SDK")
 				}
 			}
+			if rootCmdOpts.metrics && metricsServer != nil {
```

Comment:
I guess you don't need to check for the cmd opts because `metricsServer` would be `nil` if the flag is `false`.